From 717ce7ac9c2eeeee9b02ddf9310f72fa67b6600f Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 22 Sep 2020 10:09:35 +0200
Subject: [PATCH 218/529] ARM: dts: stm32: add RTC LSCO support on
 stm32mp157x-dk2

Add LSCO support on stm32mp157c-dk2 board. LSCO output is mapped on
RTC_OUT2_RMP (PI8), directly routed on LPO_IN pin of Wifi/Bluetooth
module.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I6ab004a7231b47c7590e0a6529b10f08f3cf6efe
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/278487
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/278488
Tested-by: Valentin CARON <valentin.caron@foss.st.com>
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/stm32mp15-pinctrl.dtsi | 6 ++++++
 arch/arm/boot/dts/stm32mp157c-dk2.dts    | 7 +++++++
 arch/arm/boot/dts/stm32mp157f-dk2.dts    | 7 +++++++
 3 files changed, 20 insertions(+)

--- a/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
@@ -1411,6 +1411,12 @@
 		};
 	};
 
+	rtc_out2_rmp_pins_a: rtc-out2-rmp-pins-0 {
+		pins {
+			pinmux = <STM32_PINMUX('I', 8, ANALOG)>; /* RTC_OUT2_RMP */
+		};
+	};
+
 	sai2a_pins_a: sai2a-0 {
 		pins {
 			pinmux = <STM32_PINMUX('I', 5, AF10)>, /* SAI2_SCK_A */
--- a/arch/arm/boot/dts/stm32mp157c-dk2.dts
+++ b/arch/arm/boot/dts/stm32mp157c-dk2.dts
@@ -12,6 +12,7 @@
 #include "stm32mp15xxac-pinctrl.dtsi"
 #include "stm32mp15xx-dkx.dtsi"
 #include "stm32mp157c-dk2-scmi.dtsi"
+#include <dt-bindings/rtc/rtc-stm32.h>
 
 / {
 	model = "STMicroelectronics STM32MP157C-DK2 Discovery Board";
@@ -90,6 +91,12 @@
 	};
 };
 
+&rtc {
+	st,lsco = <RTC_OUT2_RMP>;
+	pinctrl-0 = <&rtc_out2_rmp_pins_a>;
+	pinctrl-names = "default";
+};
+
 &usart2 {
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&usart2_pins_c>;
--- a/arch/arm/boot/dts/stm32mp157f-dk2.dts
+++ b/arch/arm/boot/dts/stm32mp157f-dk2.dts
@@ -12,6 +12,7 @@
 #include "stm32mp15xxac-pinctrl.dtsi"
 #include "stm32mp15xx-dkx.dtsi"
 #include "stm32mp157c-dk2-scmi.dtsi"
+#include <dt-bindings/rtc/rtc-stm32.h>
 
 / {
 	model = "STMicroelectronics STM32MP157F-DK2 Discovery Board";
@@ -90,6 +91,12 @@
 	};
 };
 
+&rtc {
+	st,lsco = <RTC_OUT2_RMP>;
+	pinctrl-0 = <&rtc_out2_rmp_pins_a>;
+	pinctrl-names = "default";
+};
+
 &usart2 {
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&usart2_pins_c>;
